lib/src/main/java/graphql/nadel/engine/transform/hydration/NadelHydrationTransform.kt

```diff
@@ -296,4 +393,73 @@ internal class NadelHydrationTransform(
                 it.condition.evaluate(node?.value)
             }
     }
+
+    private fun isDeferred(
+        executionContext: NadelExecutionContext,
+        executionBlueprint: NadelOverallExecutionBlueprint,
+        overallField: ExecutableNormalizedField,
+    ): Boolean {
+        // Disable defer in nested hydration
+        if (executionContext.hydrationDetails != null) {
+            return false
+        }
+
+        return if (executionContext.hints.deferSupport() && overallField.deferredExecutions.isNotEmpty()) {
+            // We currently don't support defer if the hydration is inside a List
+            return !areAnyParentFieldsOutputtingLists(overallField, executionBlueprint)
+        } else {
+            false
+        }
+    }
+
+    private fun areAnyParentFieldsOutputtingLists(
```

Comment:
In _this_ PR we don't allow deferring in lists. It's been implemented in https://github.com/atlassian-labs/nadel/pull/535